src/System.Memory/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.D.cs

```diff
@@ -203,122 +203,124 @@ private static bool TryParseInt32D(ReadOnlySpan<byte> text, out int value, out i
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
                 index++;
-                answer = 10 * answer + num - '0';
+                answer = 10 * answer + digit;
 
                 // Potential overflow
                 if ((uint)index >= (uint)text.Length)
                     goto Done;
-                num = text[index];
-                if (!ParserHelpers.IsDigit(num))
+                digit = text[index] - 48u; // '0';
+                if (digit > 9)
                     goto Done;
-                if (answer > Int32.MaxValue / 10 + 1)
-                    goto FalseExit; // Overflow
-                answer = answer * 10 + num - '0';
+                if (answer > int.MaxValue / 10)
```

Comment:
Is this change needed **only** because the accumulator is now a uint rather than an int?

Reply:
The accumulator changing to uint doesn't have a bearing on this. Either way, the 10 extra values that leaked through that check get caught as overflows by the second check even after multiplying by 10. But either way, what was the purpose of the "+ 1"?

Reply:
I can't figure out why we needed the "+ 1". I tried several test cases around int.MaxValue/10 and they all work with or without the "+ 1".

```C#
//(string text, bool expectSuccess, int expectedValue, int expectedConsumed)
[InlineData("214748364", true, 214748364, 9)] // int.MaxValue/10
[InlineData("214748363", true, 214748363, 9)]
[InlineData("214748362", true, 214748362, 9)]
[InlineData("214748361", true, 214748361, 9)]
[InlineData("214748365", true, 214748365, 9)]
[InlineData("214748366", true, 214748366, 9)]
[InlineData("214748367", true, 214748367, 9)]
[InlineData("214748368", true, 214748368, 9)]
[InlineData("214748369", true, 214748369, 9)]
[InlineData("214748370", true, 214748370, 9)]
[InlineData("2147483648", false, 0, 0)]
[InlineData("2147483649", false, 0, 0)]
[InlineData("2147483650", false, 0, 0)]
[InlineData("2147483647", true, 2147483647, 10)]
[InlineData("2147483646", true, 2147483646, 10)]
[InlineData("2147483645", true, 2147483645, 10)]
[InlineData("2147483644", true, 2147483644, 10)]
[InlineData("2147483643", true, 2147483643, 10)]
[InlineData("2147483642", true, 2147483642, 10)]
[InlineData("2147483641", true, 2147483641, 10)]
[InlineData("2147483640", true, 2147483640, 10)]
[InlineData("2147483639", true, 2147483639, 10)]
[InlineData("-214748364", true, -214748364, 10)] // int.MinValue/10
[InlineData("-214748363", true, -214748363, 10)]
[InlineData("-214748362", true, -214748362, 10)]
[InlineData("-214748361", true, -214748361, 10)]
[InlineData("-214748365", true, -214748365, 10)]
[InlineData("-214748366", true, -214748366, 10)]
[InlineData("-214748367", true, -214748367, 10)]
[InlineData("-214748368", true, -214748368, 10)]
[InlineData("-214748369", true, -214748369, 10)]
[InlineData("-214748370", true, -214748370, 10)]
[InlineData("-2147483648", true, -2147483648, 11)]
[InlineData("-2147483647", true, -2147483647, 11)]
[InlineData("-2147483646", true, -2147483646, 11)]
[InlineData("-2147483645", true, -2147483645, 11)]
[InlineData("-2147483644", true, -2147483644, 11)]
[InlineData("-2147483643", true, -2147483643, 11)]
[InlineData("-2147483642", true, -2147483642, 11)]
[InlineData("-2147483641", true, -2147483641, 11)]
[InlineData("-2147483640", true, -2147483640, 11)]
[InlineData("-2147483639", true, -2147483639, 11)]
[InlineData("-2147483649", false, 0, 0)]
[InlineData("-2147483650", false, 0, 0)]

        [InlineData("21474836460")] // false, 0, 0
        [InlineData("21474836480")] // false, 0, 0
        [InlineData("21474836490")] // false, 0, 0
        [InlineData("21474836500")] // false, 0, 0
        [InlineData("21474836450")] // false, 0, 0
        [InlineData("21474836440")] // false, 0, 0
        [InlineData("21474836430")] // false, 0, 0
        [InlineData("21474836420")] // false, 0, 0
        [InlineData("21474836410")] // false, 0, 0
        [InlineData("21474836400")] // false, 0, 0
        [InlineData("21474836390")] // false, 0, 0
        [InlineData("-21474836460")] // false, 0, 0
        [InlineData("-21474836480")] // false, 0, 0
        [InlineData("-21474836490")] // false, 0, 0
        [InlineData("-21474836500")] // false, 0, 0
        [InlineData("-21474836450")] // false, 0, 0
        [InlineData("-21474836440")] // false, 0, 0
        [InlineData("-21474836430")] // false, 0, 0
        [InlineData("-21474836420")] // false, 0, 0
        [InlineData("-21474836410")] // false, 0, 0
        [InlineData("-21474836400")] // false, 0, 0
        [InlineData("-21474836390")] // false, 0, 0
```

It came from a suggestion from Jan. Any ideas, Jan?
https://github.com/dotnet/corefxlab/pull/1616#issuecomment-310790392
cc @jkotas

Reply:
> It came from a suggestion from Jan. Any ideas, Jan?

Looks like oversight.